packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -634,4 +724,23 @@ - (void)setMixWithOthers:(FLTMixWithOthersMessage *)input
   }
 }
 
+- (void)preparePictureInPicture:(FLTPreparePictureInPictureMessage *)input
+                          error:(FlutterError **)error {
+  FLTVideoPlayer *player = self.playersByTextureId[input.textureId];
```

Comment:
I recommend checking to see if `UIBackgroundModes` is set in the bundle Info.plist, and to send back an error (throw an exception? @stuartmorgan what's our usual behavior for developer error in plugins?) with instructions on how to set that.

Reply:
Populating the output `NSError*` is the mechanism for Pigeon APIs in ObjC.

Reply:
@stuartmorgan is there an example somewhere on how I do that?

Reply:
Something like:
```objc
- (BOOL)doesInfoPlistSupportPictureInPicture:(FlutterError **)error {
  NSArray *backgroundModes = [NSBundle.mainBundle objectForInfoDictionaryKey:@"UIBackgroundModes"];
  if (![backgroundModes isKindOfClass:[NSArray class]] ||
      ![backgroundModes containsObject:@"audio"]) {
    *error = [FlutterError errorWithCode:@"video_player"
                                 message:@"missing audio UIBackgroundModes audio in Info.plist"
                                 details:nil];
    return NO;
  }
  return YES;
}

- (void)startPictureInPicture:(FLTStartPictureInPictureMessage *)input
                        error:(FlutterError **)error {
  if (![self doesInfoPlistSupportPictureInPicture:error]) {
    return;
  }
  FLTVideoPlayer *player = self.playersByTextureId[input.textureId];
  [player startOrStopPictureInPicture:YES];
}

- (void)stopPictureInPicture:(FLTStopPictureInPictureMessage *)input error:(FlutterError **)error {
  if (![self doesInfoPlistSupportPictureInPicture:error]) {
    return;
  }
  FLTVideoPlayer *player = self.playersByTextureId[input.textureId];
  [player startOrStopPictureInPicture:NO];
}
```